pkg/config/config_test.go

```diff
@@ -344,6 +345,30 @@ func (s *MySuite) TestDeploymentName(c *C) {
 	c.Check(errors.As(err, &e), Equals, true)
 }
 
+func (s *MySuite) TestCheckBlueprintName(c *C) {
+	dc := getDeploymentConfigForTest()
+	var e *BlueprintNameError
+
+	// Is blueprint_name a valid string?
+	err := dc.Config.checkBlueprintName()
+	c.Assert(err, IsNil)
+
+	// Is blueprint_name an empty string?
+	dc.Config.BlueprintName = ""
+	err = dc.Config.checkBlueprintName()
+	c.Check(errors.As(err, &e), Equals, true)
+
+	// Is blueprint_name longer than 63 characters?
+	dc.Config.BlueprintName = "blueprint-name-blueprint-name-blueprint-name-blueprint-name-0123"
+	err = dc.Config.checkBlueprintName()
+	c.Check(errors.As(err, &e), Equals, true)
+
+	// Does deployment_name contain special characters other than dashes or underscores?
```

Comment:
A couple more tests you could include:
* Test with an underscore (succeed)
* Test with a capital letter (fail)